phac-nml/irida-next
# Add metadata

## What does this PR do and why?
This PR allows users to add metadata to a sample through the website UI.

## Screenshots or screen recordings
Add metadata:
![image](https://github.com/phac-nml/irida-next/assets/82407232/bb475d1d-00f8-4121-8af1-bb44add88e33)
![image](https://github.com/phac-nml/irida-next/assets/82407232/bd57d4ad-f44f-4cbd-a262-8856d1feaaa0)


Add metadata with keys that already exist:
![image](https://github.com/phac-nml/irida-next/assets/82407232/06dd376a-07e8-4d16-a1e2-ba35b96d4936)
![image](https://github.com/phac-nml/irida-next/assets/82407232/9a9c5793-744f-41e0-9dc5-44e80757dfad)

Add both new and existing metadata:
![image](https://github.com/phac-nml/irida-next/assets/82407232/83326e08-4e2a-4365-9440-5eaaa3cc0a9f)
![image](https://github.com/phac-nml/irida-next/assets/82407232/2f8bda6c-57ff-4588-86da-eff8d53499a1)

Add metadata after deleting fields on form:
[Screencast from 2024-02-12 09:37:35 AM.webm](https://github.com/phac-nml/irida-next/assets/82407232/e2ccfc05-4a46-4afe-8694-f7fe535ae7f1)

## How to set up and validate locally
1. Navigate to a sample's metadata page
2. Add metadata and ensure they properly appear on the metadata table
3. Ensure adding metadata with keys that already exist are not added.
5. Fill out multiple form fields and then delete some fields. Ensure the deleted fields are not submitted and added to metadata, and only the remaining fields are added.
6. In the add metadata modal with only one field, fill out the metadata field, and then delete the field. Ensure the field still exists but is once again blank.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
